src/ext-test/java/org/opentripplanner/ext/emissions/EmissionsTest.java

```diff
@@ -0,0 +1,188 @@
+    var stopThree = TransitModelForTest.stopForTest("1:stop1", 62, 25);
+    var stopPattern = TransitModelForTest.stopPattern(stopOne, stopTwo, stopThree);
+    var route = TransitModelForTest.route(id("1")).build();
+    List<Leg> legs = new ArrayList<>();
+    var pattern = TransitModelForTest.tripPattern("1", route).withStopPattern(stopPattern).build();
+    var stoptime = new StopTime();
+    var stoptimes = new ArrayList<StopTime>();
+    stoptimes.add(stoptime);
+    var trip = Trip
+      .of(FeedScopedId.parse("FOO:BAR"))
+      .withMode(TransitMode.BUS)
+      .withRoute(route)
+      .build();
+    var leg = new ScheduledTransitLeg(
+      new TripTimes(trip, stoptimes, new Deduplicator()),
+      pattern,
+      0,
+      2,
+      TIME,
+      TIME.plusMinutes(10),
+      TIME.toLocalDate(),
+      ZoneIds.BERLIN,
+      null,
+      null,
+      100,
+      null
+    );
+    legs.add(leg);
+    Itinerary i = new Itinerary(legs);
+    assertEquals(2223.902, emissionsFilter.getEmissionsForItinerary(i, EmissionType.CO2).get());
+  }
+
+  @Test
+  void testGetEmissionsForCarRoute() {
+    List<Leg> legs = new ArrayList<>();
+    var leg = StreetLeg
+      .create()
+      .withMode(TraverseMode.CAR)
+      .withDistanceMeters(214.4)
+      .withStartTime(TIME)
+      .withEndTime(TIME.plus(1, ChronoUnit.HOURS))
+      .build();
+    legs.add(leg);
+    Itinerary i = new Itinerary(legs);
+    assertEquals(28.0864, emissionsFilter.getEmissionsForItinerary(i, EmissionType.CO2).get());
+  }
+
+  @Test
+  void testNoEmissionsForFeedWithoutEmissionsConfigured() {
+    Map<FeedScopedId, Double> emissions = new HashMap<>();
+    emissions.put(new FeedScopedId("G", "1"), (0.12 / 12));
+    EmissionsDataModel emissionsDataModel = new EmissionsDataModel();
+    emissionsDataModel.setCo2Emissions(emissions);
+    emissionsDataModel.setCarAvgCo2PerMeter(0.131);
+
+    this.eService = new EmissionsDataService(emissionsDataModel);
+    this.emissionsFilter = new EmissionsFilter(this.eService);
+
+    var route = TransitModelForTest.route(id("1")).withAgency(subject).build();
+    List<Leg> legs = new ArrayList<>();
+    var pattern = TransitModelForTest
+      .tripPattern("1", route)
+      .withStopPattern(TransitModelForTest.stopPattern(3))
+      .build();
+    var stoptime = new StopTime();
+    var stoptimes = new ArrayList<StopTime>();
+    stoptimes.add(stoptime);
+    var trip = Trip
+      .of(FeedScopedId.parse("FOO:BAR"))
+      .withMode(TransitMode.BUS)
+      .withRoute(route)
+      .build();
+    var leg = new ScheduledTransitLeg(
+      new TripTimes(trip, stoptimes, new Deduplicator()),
+      pattern,
+      0,
+      2,
+      TIME,
+      TIME.plusMinutes(10),
+      TIME.toLocalDate(),
+      ZoneIds.BERLIN,
+      null,
+      null,
+      100,
+      null
+    );
+    legs.add(leg);
+    Itinerary i = new Itinerary(legs);
+    assertThat(emissionsFilter.getEmissionsForItinerary(i, EmissionType.CO2)).isEmpty();
```

Comment:
```suggestion
    assertTrue(emissionsFilter.getEmissionsForItinerary(i, EmissionType.CO2).isEmpty());
```

Reply:
Seems like we have never used this assertThat method and I doubt we want to introduce it here.